src/Components/CopyToClipboard.jsx

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect } from "react";
+
+function CopyToClipboard({ hexCode }) {
+  const [isCopied, setIsCopied] = useState(false);
+
+  const handleCopyClick = async () => {
+    try {
+      await navigator.clipboard.writeText(hexCode);
+      setIsCopied(true);
+    } catch (error) {
+      console.error("Failed to copy: ", error);
+    }
+  };
+
+  useEffect(() => {
+    if (isCopied) {
+      const timer = setTimeout(() => setIsCopied(false), 3000);
+      return () => clearTimeout(timer);
+    }
+  }, [isCopied]);
+
+  return (
+    <div>
+      <button className="btn_copy" onClick={handleCopyClick}>
+        Copy
+      </button>
+      {isCopied && <p className="copy_confirmation">Color copied!</p>}
+    </div>
```

Comment:
i have one small suggestion :)
you can make here just one button element like this:
    <button onClick={handleCopyClick}>
      {isCopied ? "SUCCESSFULLY COPIED!" : "COPY"}
    </button>
  );
  
  it has the same function and it is shorter :)

Reply:
Yes, thank you, I’ll implement it that way. 🙌🏼